operations/commit_queue.go

```diff
@@ -117,3 +178,124 @@ func deleteCommitQueueItem(ctx context.Context, client client.Communicator, proj
 
 	return nil
 }
+
+type mergeParams struct {
+	projectID   string
+	ref         string
+	id          string
+	pause       bool
+	description string
+	skipConfirm bool
+	large       bool
+}
+
+func (p *mergeParams) mergeBranch(ctx context.Context, confPath string) error {
+	conf, err := NewClientSettings(confPath)
+	if err != nil {
+		return errors.Wrap(err, "problem loading configuration")
+	}
+
+	ac, _, err := conf.getLegacyClients()
+	if err != nil {
+		return errors.Wrap(err, "problem accessing evergreen service")
+	}
+
+	client := conf.GetRestCommunicator(ctx)
+	if err != nil {
+		return errors.Wrap(err, "problem accessing evergreen service")
+	}
+
+	if p.id == "" {
+		if p.projectID == "" {
+			p.projectID = conf.FindDefaultProject()
+		} else {
+			if conf.FindDefaultProject() == "" &&
+				!p.skipConfirm && confirm(fmt.Sprintf("Make %v your default project?", p.projectID), true) {
+				conf.SetDefaultProject(p.projectID)
+				if err := conf.Write(""); err != nil {
+					grip.Warningf("warning - failed to set default project: %v\n", err)
+				}
+			}
+		}
+		if p.projectID == "" {
+			return errors.New("no project identifier provided")
+		}
+		projectRef, err := ac.GetProjectRef(p.projectID)
+		if err != nil {
+			return err
+		}
+		diffData, err := getFeaturePatchInfo(projectRef.Branch, p.ref)
+		if err != nil {
+			return errors.Wrap(err, "can't generate patches")
+		}
+
+		if p.description == "" && !p.skipConfirm {
+			p.description = prompt("Enter a description for this patch (optional):")
+		}
+		if p.description == "" {
+			p.description, err = gitCmd("rev-parse", "--abbrev-ref", p.ref)
+			if err != nil {
+				return errors.Wrapf(err, "can't get branch name for ref %s", p.ref)
+			}
+		}
+
+		patchParams := &patchParams{
+			Project:     p.projectID,
+			SkipConfirm: p.skipConfirm,
+			Description: p.description,
+			Large:       p.large,
+			Alias:       commitQueueAlias,
+		}
+
+		patch, err := patchParams.createPatch(ac, conf, diffData)
+		if err != nil {
+			return err
+		}
+		if patch == nil {
+			return nil
+		}
+		p.id = patch.Id.Hex()
+	}
+
+	if !p.pause {
+		position, err := client.EnqueueItem(ctx, p.id)
+		if err != nil {
+			return err
+		}
+		grip.Infof("Queue position is %d", position)
+	}
+
+	return nil
+}
+
+func getFeaturePatchInfo(projectBranch, ref string) (*localDiff, error) {
+	upstream := projectBranch + "@{upstream}"
```

Comment:
assume a local branch exists with 
- the same name as the project branch, and
- tracking the project branch

This is reasonable considering we already make this assumption for patches.